File: Greedy/55_jump_game/55.cpp



class Solution {
public:
    bool canJump(vector<int>& nums) {
        int good_position = nums.size() - 1;
        for (int i = nums.size() - 1; i >= 0; i--) {
            if (nums[i] + i >= good_position) {
                good_position = i;
            }
        }
        return good_position == 0;
    }
};

File: Greedy/55_jump_game/55.rs


impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let mut good_position: usize = nums.len() - 1;
        for i in (0..nums.len()).rev() {
            if i + nums[i] as usize >= good_position {
                good_position = i;
            }
        }
        return good_position == 0;
    }
}